openedx/core/djangoapps/content/learning_sequences/views.py

```diff
@@ -213,7 +213,12 @@ def _determine_user(self, request, course_key: CourseKey) -> types.User:
 
         target_username = request.GET.get("user")
         if target_username is not None:
-            return self._get_target_user(request, course_key, has_staff_access, target_username)
+            target_user = self._get_target_user(request, course_key, has_staff_access, target_username)
+            # Just like in masquerading, set real_user so that the
+            # SafeSessions middleware can see that the user didn't
+            # change unexpectedly.
+            target_user.real_user = request.user
+            return target_user
```

Comment:
Author note: I don't know why this wasn't causing failures in production, either. But... I _think_ it might not be enabled/in use yet? Haven't checked in detail.